### PR TITLE
DAOS-5824 control: move control api print functions into dmg

### PR DESCRIPTION
**Pull request by tanabarr**
Control API package has print functions to display human readable
representations of results maps returned from requests made through
the API.  Move these into the pretty package which exists in the
dmg tool directory as the display translation of the API results
should be the responsibility of the API consumer.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Review by mjmac — APPROVED** [review]
LGTM, thanks for tackling this. It needed to happen.

File: src/control/cmd/dmg/pretty/printers.go [hunk 1]
```diff
@@ -23,3 +23,138 @@
...
+// PrintWithHostPorts enables display of host ports in output.
+func PrintWithHostPorts() PrintConfigOption {
+	return func(cfg *PrintConfig) {
+		cfg.ShowHostPorts = true
+	}
+}
+
+// GetPrintConfig is a helper that returns a format configuration
+// for a format function.
+func GetPrintConfig(opts ...PrintConfigOption) *PrintConfig {
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/cmd/dmg/pretty/printers.go:81]: addressing in https://github.com/daos-stack/daos/pull/3589

File: src/control/cmd/dmg/pretty/printers_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+package pretty
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/daos-stack/daos/src/control/lib/control"
+	"github.com/google/go-cmp/cmp"
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/cmd/dmg/pretty/printers_test.go:32]: addressing in https://github.com/daos-stack/daos/pull/3589

File: src/control/cmd/dmg/pretty/printers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/cmd/dmg/pretty/printers.go:81]: done

File: src/control/cmd/dmg/pretty/printers_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/cmd/dmg/pretty/printers_test.go:32]: done

File: src/control/cmd/dmg/pretty/storage_scm_test.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package pretty
```
Comment (kjacque) [inline src/control/cmd/dmg/pretty/storage_scm_test.go:24]: Were there no existing tests for the SCM print functions? I'm okay with addressing them in a follow-on, but seems like we should have tests.

File: src/control/cmd/dmg/storage_test.go [hunk 4]
```diff
@@ -1,330 +0,0 @@
...
-	"github.com/daos-stack/daos/src/control/common"
-	ctlpb "github.com/daos-stack/daos/src/control/common/proto/ctl"
-	"github.com/daos-stack/daos/src/control/lib/control"
-	"github.com/daos-stack/daos/src/control/logging"
-	"github.com/daos-stack/daos/src/control/system"
-)
-
-func TestStorageCommands(t *testing.T) {
-	storageFormatReq := &control.StorageFormatReq{Reformat: true}
-	storageFormatReq.SetHostList([]string{})
```
Comment (kjacque) [inline src/control/cmd/dmg/storage_test.go:42]: Might have missed something, but is this test duplicated somewhere else? Moved? Seems like we should keep this one at least.

File: src/control/cmd/dmg/pretty/storage_scm_test.go (hunk 3 above)
Comment (tanabarr) [reply src/control/cmd/dmg/pretty/storage_scm_test.go:24]: yes , they seem to be missing, added this as placeholder

File: src/control/cmd/dmg/storage_test.go (hunk 4 above)
Comment (tanabarr) [reply src/control/cmd/dmg/storage_test.go:42]: whoops, adding back as part of https://github.com/daos-stack/daos/pull/3589 , good catch . thanks

**Review by kjacque — APPROVED** [review]

**Pull request merged**